### PR TITLE
Add stb/cci.20210713

### LINKED ISSUE #6921: [request] stb/20210818
### Package Details
  * Package Name/Version: **stb/20210818**

There has been +1800 commits added to stb since Feb 2 of 2020, I greatly suggest updating it.

The above mentioned version is newly released by the upstream project and not yet available as a recipe. Please add this version.

### PR DESCRIPTION
**Pull request by TheClonerx**
Specify library name and version:  **stb/cci.20210713**

This closes #6921.

---

- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [X] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [X] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/stb/all/conanfile.py [hunk 1]
```diff
@@ -1,27 +1,46 @@
...
+    default_options = {
+        "with_deprecated": True
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.version < "20210713":
```
Comment (madebr) [inline recipes/stb/all/conanfile.py:28]:
```suggestion
        if tools.Version(self.version) < "20210713":
```

File: recipes/stb/all/conanfile.py [hunk 2]
```diff
@@ -1,27 +1,46 @@
 from conans import ConanFile, tools
 import os
 
+required_conan_version = ">=1.33.0"
+
 class StbConan(ConanFile):
```
Comment (madebr) [inline recipes/stb/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.33.0"


class StbConan(ConanFile):
```

File: recipes/stb/all/conandata.yml [hunk 3]
```diff
@@ -2,3 +2,6 @@ sources:
   "20200203":
     url: "https://github.com/nothings/stb/archive/0224a44a10564a214595797b4c88323f79a5f934.zip"
     sha256: "5dd9cf8a9a8c0f253fad3e9904923b2dfc740eddbc86415c3f76152bd534f23a"
+  "20210713":
```
Comment (madebr) [inline recipes/stb/all/conandata.yml:5]: We have started to use a `cci.YYYYMMDD` format.
But I don't think `tools.Version` is compatible with that.

File: recipes/stb/all/conanfile.py [hunk 4]
```diff
@@ -1,27 +1,46 @@
...
-        extracted_dir = self.name + "-" + commit
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 
     def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         self.copy("*.h", src=self._source_subfolder, dst="include")
         self.copy("stb_vorbis.c", src=self._source_subfolder, dst="include")
         tools.rmdir(os.path.join(self.package_folder, "include", "tests"))
+        if self.version >= "20210713":
```
Comment (uilianries) [inline recipes/stb/all/conanfile.py:39]:
```suggestion
        if tools.Version(self.version) >= "20210713":
```

File: recipes/stb/all/conandata.yml (hunk 3 above)
Comment (uilianries) [reply recipes/stb/all/conandata.yml:5]: True, it's incompatible.

Comment (TheClonerx) [reply recipes/stb/all/conandata.yml:5]: Should I also change it for version `20200203`?

Comment (madebr) [reply recipes/stb/all/conandata.yml:5]: Please do. Older versions remain available on conan-center.
To tackle the version comparison, perhaps create a helper method `self._version` and use that for version comparison
```python
def _version(self):
    return str(self.version)[4:]
```

Alternatively, don't change the version and use:
```
def _version(self):
    import re
    return re.match(r"^(?:cci\.|)(.*)", str(self.version)).group(1)
```

File: recipes/stb/all/conanfile.py [hunk 5]
```diff
@@ -1,27 +1,53 @@
...
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
 
     def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         self.copy("*.h", src=self._source_subfolder, dst="include")
         self.copy("stb_vorbis.c", src=self._source_subfolder, dst="include")
         tools.rmdir(os.path.join(self.package_folder, "include", "tests"))
+        if tools.Version(self._version) >= "20210713":
+            tools.rmdir(os.path.join(self.package_folder, "include", "deprecated"))
```
Comment (madebr) [inline recipes/stb/all/conanfile.py:47]: I think you can do this unconditionally.
It won't fail if the directory does not exist.

Comment (Croydon) [reply recipes/stb/all/conanfile.py:47]: Personally, I prefer to have things like this directly connected to recipe version. Helps me to understand immediately what behaviour / files to except in specific versions

Comment (TheClonerx) [reply recipes/stb/all/conanfile.py:47]: I did this with the "Better safe than sorry" mentality, but also wanting to explicitly express differences between the versions.

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**